packages/integration/src/azure/config.ts

```diff
@@ -71,40 +72,73 @@ export type ClientSecret = {
  * Authenticate using a managed identity available at the deployment environment.
  * @public
  */
-export type ManagedIdentity = {
+export type ManagedIdentityCredential = AzureIntegrationCredentialConfigBase & {
   /**
    * The clientId
    */
   clientId: string;
 };
 
+export type PersonalAccessTokenCredential =
+  AzureIntegrationCredentialConfigBase & {
+    token: string;
+  };
+
 /**
- * Credential used to authenticate to Azure Active Directory.
+ * Credential used to authenticate to Azure DevOps.
  * @public
  */
-export type AzureCredential = ClientSecret | ManagedIdentity;
-export const isServicePrincipal = (
-  credential: Partial<AzureCredential>,
-): credential is ClientSecret => {
-  const clientSecretCredential = credential as ClientSecret;
+export type AzureIntegrationCredentialConfig =
+  | ClientSecretCredential
+  | ManagedIdentityCredential
+  | PersonalAccessTokenCredential;
+
+const getConfigKeys = (credential: Partial<AzureIntegrationCredentialConfig>) =>
+  Object.keys(credential)
+    .filter(key => key !== 'organizations')
+    .filter(
+      key =>
+        credential[key as keyof AzureIntegrationCredentialConfig] !== undefined,
+    );
+
+export const isClientSecret = (
+  credential: Partial<AzureIntegrationCredentialConfig>,
+): credential is ClientSecretCredential => {
+  const clientSecretCredential = credential as ClientSecretCredential;
 
   return (
-    Object.keys(credential).length === 3 &&
+    getConfigKeys(credential).length === 3 &&
```

Comment:
Do we really need to count keys here? 😅 is it not enough to check for the presence of the 3 props?

Reply:
Probably not. The reason I did this was to prevent configs like this:
```yaml
host: dev.azure.com
credentials: 
  - token: token
    clientId: id
    clientSecret: secret
    tenantId: tenant
```` 
This config would match `AzureClientSecretCredential`, `AzureManagedIdentityCredential`, and `PersonalAccessTokenCredential`. But I can probably prevent this without counting keys.

Reply:
Fixed, I refactored the type predicates to check the specific properties instead.